### ISSUE TITLE
Can you please provide the code for drawing Figure 5?

### ISSUE THREAD

**Issue by YuMJie** [LTH14/targeted-supcon#6]
Thanks for your great work! Figure 5 in your paper is also impressive, could you please provide the code for drawing this figure?

**Comment by JayC1208** [comment]
I am also interested in that code. 
Could you provide the code or give information about the method/python library or other repos that used similar method?

**Comment by LTH14** [comment]
We don't have the plan to release the code for plotting figure 5 as it requires quite a bit of cleaning. However, we simply use matplotlib scatter with alpha=0.1 to plot the features. One thing to notice is that, for better visualization, to plot figure 5 we set the output dimension of the MLP to 2. As a result, the normalized features used to compute the loss naturally fall on a 2-d circle.

**Comment by RollingSong** [comment]
In fact, we can do this simply by using t_SNE without extra trainning or fine-tune. 
(1) use sklearn.manifold.TSNE, change the channel of feature vector from your output dim to 2, just same as many work do.
(2) use np.linalg.norm or some ways to normlize our vector2.
(3) use plt.scatter with alpha=0.1 or smaller 
That's all. 
This image is drawn by my own method with defaut color，it has a similar visual effect. Certainly much uglier than the picture in this work~~
![t_SNE with norm](https://github.com/user-attachments/assets/9d311907-d388-4e22-8125-ca65a2105e23)
